lib/SparqlEndpointFetcher.ts

```diff
@@ -206,16 +208,24 @@ export class SparqlEndpointFetcher {
     }
 
     // Initiate request
-    let body: URLSearchParams | undefined;
+    let body: URLSearchParams | string | undefined;
     const headers: Headers = new Headers(this.defaultHeaders);
     headers.append('Accept', acceptHeader);
 
     if (method === 'POST') {
-      headers.append('Content-Type', 'application/x-www-form-urlencoded');
-      body = new URLSearchParams();
-      body.set('query', query);
-      for (const [ key, value ] of this.additionalUrlParams.entries()) {
-        body.set(key, value);
+      if (this.forceDirectPost) {
+        headers.append('Content-Type', 'application/sparql-query');
+        body = query;
+        if (this.additionalUrlParams.toString().length > 0) {
```

Comment:
An URLSearchParam is not the same as a query string param, but I get why you're doing this.
We should make sure that we document this behaviour in the docs of `additionalUrlParams`.